---
title: "lab14.0"
format: pdf
editor: visual
author: Ilyas Darif
date: 2/20/2025
toc: true 
---

## Background

The data for for hands-on session comes from GEO entry: GSE37704, which is associated with the following publication:

> Trapnell C, Hendrickson DG, Sauvageau M, Goff L et al. "Differential analysis of gene regulation at transcript resolution with RNA-seq". Nat Biotechnol 2013 Jan;31(1):46-53. PMID: 23222703

The authors report on differential analysis of lung fibroblasts in response to loss of the developmental transcription factor HOXA1. Their results and others indicate that HOXA1 is required for lung fibroblast and HeLa cell cycle progression. In particular their analysis show that "loss of HOXA1 results in significant expression level changes in thousands of individual transcripts, along with isoform switching events in key regulators of the cell cycle". For our session we have used their Sailfish gene-level estimated counts and hence are restricted to protein-coding genes only.

## Data Import

```{r}
counts <- read.csv("GSE37704_featurecounts.csv", row.names=1)
colData <- read.csv("GSE37704_metadata.csv")
```

## Inspect and tidy data

Does the `counts` columns mathc the `colData` rows?

```{r}
head(counts)
```

```{r}
colData$id
```

```{r}
colnames(counts)
```

```{r}
countData <- counts[,-1]
head(countData)
```

check for matching count data and colData

```{r}
colnames(countData) == colData
```

> Q1. How many genes in total

```{r}
nrow(countData)
```

> Q2 Filter to remove zero count genes (rows where there are zero counts in all columns). How many genes are left?

```{r}
to.keep.inds <- rowSums(countData) > 0
```

```{r}
new.counts <- countData[to.keep.inds,]
```

```{r}
nrow(new.counts)
```

## Setup for DESeq

```{r}
#| message: false
library(DESeq2)
```

Setup input object for DESeq

```{r}
dds <- DESeqDataSetFromMatrix(countData = new.counts,
                              colData = colData,
                              design = ~condition)
```

## Run DESeq

```{r}
dds <- DESeq(dds)
res <- results(dds)
```

```{r}
head(res)
```

## Volcano plot of results

```{r}
library(ggplot2)
```

```{r}
# Setup our custom point color vector 
mycols <- rep("green", nrow(res))
mycols[ abs(res$log2FoldChange) > 2 ]  <- "red" 

inds <- (res$padj < 0.05) & (abs(res$log2FoldChange) > 2 )
mycols[ inds ] <- "blue"

# Volcano plot with custom colors 
plot( res$log2FoldChange,  -log(res$padj), 
 col=mycols, ylab="-Log(P-value)", xlab="Log2(FoldChange)" )

# Cut-off lines
abline(v=c(-2,2), col="black", lty=2)
abline(h=-log(0.1), col="gray", lty=2)
```


```{r}
ggplot(res) + 
  aes(log2FoldChange, -log2(padj)) + 
  geom_point() +
  geom_vline(xintercept = c(-2,2), col="red") +
  geom_hline(yintercept = -log(0.005), col="red")

```

## Gene annotation

```{r}
library(AnnotationDbi)
library(org.Hs.eg.db)
```

```{r}
columns(org.Hs.eg.db)
```

Add gene SYMBOL and ENTREZID

```{r}
res$symbol = mapIds(org.Hs.eg.db,
                    keys=rownames(res), 
                    keytype="ENSEMBL",
                    column="SYMBOL")

res$entrez = mapIds(org.Hs.eg.db,
                    keys=rownames(res),
                    keytype="ENSEMBL",
                    column="ENTREZID")


head(res, 10)
```

## Pathway analysis

```{r}
library(gage)
library(gageData)
library(pathview)
```

Input vector for `gage()`

```{r}
foldchanges = res$log2FoldChange
names(foldchanges) = res$entrez
```

Load up the KEGG genesets

```{r}
data(kegg.sets.hs)
```

Run pathway analysis

```{r}
keggres = gage(foldchanges, gsets=kegg.sets.hs)
```

```{r}
head(keggres$less)
```

Cell cylce figure

```{r}
pathview(foldchanges, pathway.id = "hsa04110")
```

![](hsa04110.png)

DNA Replication Figure

```{r}
pathview(foldchanges, pathway.id = "hsa03030")
```

## Gene Ontology analysis

Run pathway analysis with GO

```{r}
data(go.sets.hs)
data(go.subs.hs)


# Focus on Biological Process subset of GO
gobpsets = go.sets.hs[go.subs.hs$BP]

gobpres = gage(foldchanges, gsets = gobpsets, same.dir=TRUE)

lapply(gobpres, head)
```

```{r}
head(keggres$greater)
```

DNA Replication Figure

```{r}
pathview(foldchanges, pathway.id = "hsa03030")
```

![](hsa03030.png)

RNA transport figure

```{r}
pathview(foldchanges, pathway.id = "hsa03013")
```

![](hsa03013.png)

Amoebiasis figure

```{r}
pathview(foldchanges, pathway.id = "hsa05146")
```

![](hsa05146.png)

Rheumatoid arthritis figure

```{r}
pathview(foldchanges, pathway.id = "hsa05323")
```

![](hsa05323.png)

```{r}
sig_genes <- res[res$padj <= 0.05 & !is.na(res$padj), "symbol"]
print(paste("Total number of significant genes:", length(sig_genes)))
```

```{r}
write.table(sig_genes, file="significant_genes.txt", row.names=FALSE, col.names=FALSE, quote=FALSE)
```

> What pathway has the most significant “Entities p-value”? Do the most significant pathways listed match your previous KEGG results? What factors could cause differences between the two methods? - Cell Cycle it does match the kegg results top value matches the previous go looks at gen function at a standard basic is just a basic way of looking at the pathways kegg looks at gene interaction through a biological pathway in a more complex way
